### ISSUE TITLE
ng generate component fails to find the NgModule in a multi-app CLI project

### ISSUE THREAD

**Issue by johnpapa** [angular/angular-cli#9771]
### Versions
```
Angular CLI: 1.7.1
Node: 8.9.4
OS: darwin x64
Angular: 5.2.6
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cdk: 5.2.2
@angular/cli: 1.7.1
@angular/material: 5.2.2
@angular-devkit/build-optimizer: 0.3.2
@angular-devkit/core: 0.3.2
@angular-devkit/schematics: 0.3.2
@ngtools/json-schema: 1.2.0
@ngtools/webpack: 1.10.1
@schematics/angular: 0.3.2
@schematics/package-update: 0.3.2
typescript: 2.5.3
webpack: 3.11.0
```

### Repro steps

1. Create a multi-app project with the CLI
2. name one app1 and the other app2 in `angular-cli.json`
3. type `ng g c foo -a app1`

### Observed behavior

An error occurs saying it cannot find the module to declare the component.

We can *try* to work around this by typing `ng g c foo -a 1-routing-end --skip-import` and then manually declaring the component. But  this doesnt work as intended as it puts the files in the `app` folder, when I expected it to put them in the `app1` folder.

```
ng g c foo -a app1 --skip-import --spec false -dr
  create src/client/app/foo/foo.component.scss (0 bytes)
  create src/client/app/foo/foo.component.html (43 bytes)
  create src/client/app/foo/foo.component.ts (339 bytes)
```

### Desired behavior

`ng g c foo -a app1` should generate the `FooComponent` in `app1` and declare it in the closest module.

it should put it in the folder where the specified app is indicated.

### Mention any other details that might be useful (optional)

Adding the `-m` flag doesn't help, fully qualified path or not.

**Comment by johnpapa** [comment]
Found the solution!

I did not have `appRoot` in the `angular-cli.json` file. Thus it looks like that tells the CLI where the app's root it. So now I have this:

```
      "name": "app1",
      "outDir": "dist/app1",
      "main": "main.app1.ts",
      "appRoot": "app1",
```

and all is well.

@Brocco ... do you think it makes sense to document this in the wiki (that all of those are important)? Or did I miss it somewhere in there? Or maybe have a better error message saying the appRoot is missing.

**Comment by venturads** [comment]
try ng g c --project testapp FirstPage

**Comment by hansl** [comment]
I'm going to close this PR as it's been solved with the new workspace feature and we won't fix non-security bug in 1.7 anymore. Thanks!
